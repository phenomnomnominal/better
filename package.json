{
  "name": "@betterer/monorepo",
  "version": "0.0.0",
  "private": true,
  "author": "Craig Spence <craigspence0@gmail.com>",
  "license": "MIT",
  "description": "Betterer makes it easier to make incremental improvements to your codebase",
  "workspaces": {
    "packages": [
      "packages/*"
    ]
  },
  "scripts": {
    "api": "yarn compile && lerna run api",
    "betterer": "node ./packages/cli/./bin/betterer",
    "betterer:ci": "node ./packages/cli/./bin/betterer-ci",
    "build": "yarn compile && yarn format && yarn lint && yarn test && yarn test:api && yarn betterer",
    "build:ci": "yarn compile && yarn format && yarn lint && yarn test && yarn test:api && yarn betterer:ci",
    "build:extension": "lerna run build:extension",
    "clean": "yarn clean:tests && yarn clean:compile && yarn clean:modules",
    "clean:tests": "rimraf fixtures && rimraf reports",
    "clean:compile": "lerna exec \"rimraf dist\" && rimraf ./**/tsconfig.tsbuildinfo",
    "clean:modules": "lerna exec \"rimraf node_modules\" && rimraf node_modules && yarn",
    "commit": "git cz",
    "compile": "tsc -b packages",
    "compile:watch": "tsc -b packages -w",
    "lint": "eslint \"./packages/**/src/**/*.{js,ts}\" \"./test/**/*.{js,ts}\" \"./*.{js,ts}\" --fix",
    "format": "prettier \"**/!(*.snap)\" --write --loglevel=silent",
    "test": "yarn clean:tests && jest",
    "test:api": "node -r ts-node/register ./test/api",
    "test:debug": "yarn clean:tests && jest --runInBand --collectCoverage=false",
    "test:extension": "lerna run test:extension",
    "prepublishOnly": "yarn run build"
  },
  "engines": {
    "node": ">=12"
  },
  "types": "test/betterer-public-api.d.ts",
  "devDependencies": {
    "@betterer/cli": "^3.1.2",
    "@betterer/errors": "^3.1.1",
    "@betterer/eslint": "^3.1.2",
    "@betterer/logger": "^3.1.0",
    "@betterer/regexp": "^3.1.2",
    "@betterer/tsquery": "^3.1.2",
    "@commitlint/cli": "^11.0.0",
    "@commitlint/config-conventional": "^11.0.0",
    "@phenomnomnominal/commitlint-plugin": "^1.1.1",
    "@phenomnomnominal/worker-require": "^0.0.13",
    "@types/eslint": "^7.2.4",
    "@types/fs-extra": "^9.0.2",
    "@types/graceful-fs": "^4.1.3",
    "@types/jest": "^26.0.0",
    "@types/node": "^14.11.8",
    "@types/react": "^16.9.51",
    "@typescript-eslint/eslint-plugin": "^4.4.1",
    "@typescript-eslint/parser": "^4.4.1",
    "ansi-regex": "^5.0.0",
    "cz-conventional-changelog": "^3.3.0",
    "eslint": "^7.9.0",
    "eslint-config-prettier": "^7.0.0",
    "eslint-plugin-jest": "^24.0.1",
    "eslint-plugin-prettier": "^3.1.4",
    "husky": "^4.3.0",
    "ink": "^3.0.7",
    "jest": "^26.5.3",
    "fs-extra": "^9.0.1",
    "graceful-fs": "^4.2.3",
    "lerna": "^3.22.1",
    "npm-run-all": "^4.1.5",
    "react": "^16.13.1",
    "prettier": "^2.1.1",
    "ts-api-guardian": "^0.5.0",
    "ts-jest": "^26.3.0",
    "ts-node": "^9.0.0",
    "typescript": "^4.1.3"
  },
  "husky": {
    "hooks": {
      "pre-commit": "yarn build && git add .betterer.results",
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
    }
  },
  "config": {
    "commitizen": {
      "path": "cz-conventional-changelog"
    }
  }
}
