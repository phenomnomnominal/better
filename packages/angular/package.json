{
  "name": "@betterer/angular",
  "description": "Angular test for @betterer/betterer",
  "version": "5.4.0",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "author": "Craig Spence <craigspence0@gmail.com>",
  "homepage": "https://phenomnomnominal.github.io/betterer",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/phenomnomnominal/betterer.git"
  },
  "bugs": {
    "url": "https://github.com/phenomnomnominal/betterer/issues"
  },
  "scripts": {
    "compile": "tsc -b .",
    "api": "api-extractor run --local --verbose"
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "@betterer/betterer": "^5.4.0",
    "@betterer/errors": "^5.3.0",
    "tslib": "^2.3.1"
  },
  "devDependencies": {
    "@angular/common": "^12.1.0",
    "@angular/compiler": "^12.1.0",
    "@angular/compiler-cli": "^12.1.0",
    "@angular/core": "^12.1.0",
    "@angular/platform-browser": "^12.1.0",
    "@angular/platform-browser-dynamic": "^12.1.0",
    "rxjs": "^7.1.0",
    "zone.js": "^0.11.4"
  },
  "peerDependencies": {
    "@angular/compiler": "*",
    "@angular/compiler-cli": "*",
    "typescript": "*"
  }
}
