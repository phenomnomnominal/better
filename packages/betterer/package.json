{
  "name": "@betterer/betterer",
  "description": "Main engine for runing betterer tests",
  "version": "3.1.2",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "author": "Craig Spence <craigspence0@gmail.com>",
  "homepage": "https://phenomnomnominal.github.io/betterer",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/phenomnomnominal/betterer.git"
  },
  "bugs": {
    "url": "https://github.com/phenomnomnominal/betterer/issues"
  },
  "scripts": {
    "compile": "tsc -b .",
    "api": "ts-api-guardian --out ../../goldens/api/@betterer/betterer.d.ts dist/index.d.ts"
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "@betterer/constraints": "^3.1.1",
    "@betterer/errors": "^3.1.1",
    "@betterer/logger": "^3.1.1",
    "@betterer/reporter": "^3.1.2",
    "@betterer/watch-reporter": "^3.1.2",
    "@phenomnomnominal/debug": "^0.2.5",
    "callsite": "^1.0.0",
    "chokidar": "^3.3.1",
    "djb2a": "^1.2.0",
    "gitignore-globs": "^0.1.1",
    "globby": "^11.0.1",
    "lines-and-columns": "^1.1.6",
    "minimatch": "^3.0.4",
    "ts-node": "^9.0.0",
    "tslib": "^2.0.3"
  },
  "devDependencies": {
    "@types/callsite": "^1.0.30"
  }
}
