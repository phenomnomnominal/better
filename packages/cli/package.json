{
  "name": "@betterer/cli",
  "description": "betterer CLI",
  "version": "3.1.2",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "bin",
    "dist"
  ],
  "bin": {
    "betterer": "./bin/betterer"
  },
  "author": "Craig Spence <craigspence0@gmail.com>",
  "homepage": "https://phenomnomnominal.github.io/betterer",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/phenomnomnominal/betterer.git"
  },
  "bugs": {
    "url": "https://github.com/phenomnomnominal/betterer/issues"
  },
  "scripts": {
    "compile": "tsc -b .",
    "api": "ts-api-guardian --out ../../goldens/api/@betterer/cli.d.ts dist/index.d.ts"
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "@betterer/betterer": "^3.1.2",
    "@betterer/errors": "^3.1.1",
    "@phenomnomnominal/worker-require": "^0.0.13",
    "commander": "^5.1.0",
    "find-up": "^5.0.0",
    "ink": "^3.0.7",
    "react": "^16.13.1",
    "tslib": "^2.0.3"
  }
}
