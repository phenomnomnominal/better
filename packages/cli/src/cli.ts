import commander from 'commander';

import { BettererCLIArguments, BettererPackageJSON } from './types';

enum Command {
  ci = 'ci',
  start = 'start',
  init = 'init',
  watch = 'watch'
}

type CommandName = `${Command}`;

// HACK:
// It's easier to use require than to try to get `await import`
// to work right for the package.json...
/* eslint-disable @typescript-eslint/no-var-requires */
const { version } = require('../package.json') as BettererPackageJSON;

export function cliÎ”(argv: BettererCLIArguments): void {
  commander.version(version);

  commander.command(Command.ci, 'run Betterer in CI mode');
  commander.command(Command.init, 'init Betterer in a project');
  commander.command(Command.start, 'run Betterer');
  commander.command(Command.watch, 'run Betterer in watch mode');

  const args = argv.slice(0);
  const [, , command] = args;
  if (!Command[command as CommandName]) {
    args.splice(2, 0, Command.start);
  }

  commander.parse(args);
}
