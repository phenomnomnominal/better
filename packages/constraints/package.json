{
  "name": "@betterer/constraints",
  "description": "default constraints for @betterer/betterer",
  "version": "3.1.1",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "author": "Craig Spence <craigspence0@gmail.com>",
  "homepage": "https://phenomnomnominal.github.io/betterer",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/phenomnomnominal/betterer.git"
  },
  "bugs": {
    "url": "https://github.com/phenomnomnominal/betterer/issues"
  },
  "scripts": {
    "compile": "tsc -b .",
    "api": "ts-api-guardian --out ../../goldens/api/@betterer/constraints.d.ts dist/index.d.ts"
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "tslib": "^2.0.3"
  }
}
