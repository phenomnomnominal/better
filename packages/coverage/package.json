{
  "name": "@betterer/coverage",
  "description": "A betterer test which ensures that test coverage increases over time",
  "version": "5.4.0",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "author": "Lucas TreffenstÃ¤dt <lucas.treffenstaedt@check24.de>",
  "homepage": "https://phenomnomnominal.github.io/betterer",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/phenomnomnominal/betterer.git"
  },
  "bugs": {
    "url": "https://github.com/phenomnomnominal/betterer/issues"
  },
  "scripts": {
    "compile": "tsc -b .",
    "api": "api-extractor run --local --verbose"
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "@betterer/betterer": "^5.4.0",
    "@betterer/constraints": "^5.3.0",
    "@betterer/errors": "^5.3.0",
    "@betterer/logger": "^5.3.4",
    "minimatch": "^5.0.1",
    "tslib": "^2.3.1"
  },
  "devDependencies": {
    "@types/istanbul-lib-coverage": "2.0.4",
    "@types/minimatch": "^3.0.5"
  }
}
