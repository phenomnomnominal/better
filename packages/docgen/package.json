{
  "name": "doc-plugin-betterer",
  "description": "@microsoft/api-documenter plugin for @betterer/betterer",
  "version": "5.3.0",
  "license": "MIT",
  "private": true,
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "type": "commonjs",
  "files": [
    "dist"
  ],
  "author": "Craig Spence <craigspence0@gmail.com>",
  "homepage": "https://phenomnomnominal.github.io/betterer",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/phenomnomnominal/betterer.git"
  },
  "bugs": {
    "url": "https://github.com/phenomnomnominal/betterer/issues"
  },
  "scripts": {
    "compile": "tsc -b ."
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "dedent": "^0.7.0",
    "tslib": "^2.3.1"
  },
  "devDependencies": {
    "@types/dedent": "^0.7.0"
  }
}
