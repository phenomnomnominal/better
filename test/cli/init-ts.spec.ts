import { BettererPackageJSON, initŒî } from '@betterer/cli';

import { createFixture } from '../fixture';

const ARGV = ['node', './bin/betterer'];

import { version } from '../../packages/cli/package.json';

describe('betterer cli', () => {
  it('should initialise betterer in a repo with TS', async () => {
    const { cleanup, logs, paths, readFile, resolve } = await createFixture(
      'init-ts',
      {
        'package.json': `
      {
        "name": "init-ts",
        "version": "0.0.1"
      }
      `
      },
      {
        logFilters: [/üåü Initialising Betterer/]
      }
    );

    const configPath = `${paths.config}.ts`;
    const fixturePath = paths.cwd;
    const packageJSONPath = resolve('./package.json');

    await initŒî(fixturePath, ARGV);

    const packageJSON = JSON.parse(await readFile(packageJSONPath)) as BettererPackageJSON;

    expect(packageJSON.scripts.betterer).toEqual('betterer');
    expect(packageJSON.devDependencies['@betterer/cli']).toEqual(`^${version}`);
    expect(packageJSON.devDependencies['typescript']).toBeDefined();

    const config = await readFile(configPath);

    expect(config).toEqual('export default {\n  // Add tests here ‚òÄÔ∏è\n};\n');

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
