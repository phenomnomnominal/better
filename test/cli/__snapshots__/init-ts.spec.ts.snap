// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`betterer cli should initialise betterer in a repo with TS 1`] = `
Array [
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

ðŸŽ‰ Initialising Betterer (0ms): 2 tasks done!
âœ… Create test file: done!
ãƒ» created \\"<project>/fixtures/init-ts/.betterer.ts\\"!
âœ… Update package.json: done!
ãƒ» added \\"betterer\\" script to package.json file.
ãƒ» added \\"@betterer/cli\\" dependency to package.json file
ãƒ» added \\"typescript\\" dependency to package.json file",
]
`;
